### ISSUE TITLE
Should we add an option to update instead of fit when using cross_val_predict?

### ISSUE THREAD

**Issue by tuomijal** [alkaline-ml/pmdarima#339]
**Question**

I wonder if it would make sense to add an option to use [update](http://alkaline-ml.com/pmdarima/modules/generated/pmdarima.arima.ARIMA.html?highlight=update#pmdarima.arima.ARIMA.update) insted of [fit](http://alkaline-ml.com/pmdarima/modules/generated/pmdarima.arima.ARIMA.html?highlight=fit#pmdarima.arima.ARIMA.fit) when using [cross_val_predict](http://alkaline-ml.com/pmdarima/modules/generated/pmdarima.model_selection.cross_val_predict.html#pmdarima.model_selection.cross_val_predict)? 

This question is motivated by some tests included below. Tests are run on my own dataset but I will provide and executable MRE *per request*. I have a daily timeseries with 1266 observations. I split this into train set of 731 observations and test set of 535 observations. Metrics are calculated on the test set.

First we test the novel `cross_val_predict` method which uses `fit` on each iteration.
```python
# test 1
%%time

from pmdarima import model_selection, auto_arima
from tiima.metrics import PredictionResult

cv = model_selection.RollingForecastCV(h=1, step=1, initial=y_train.shape[0])
predictions = model_selection.cross_val_predict(est1, y, cv=cv, verbose=0)

n_test = predictions.shape[0]
y_pred = pd.Series(predictions, index=y.index[-n_test:])

pr_cv = PredictionResult()
pr_cv.fit(y_pred, y_test)

print(f"MAPE: {pr_cv.mean_absolute_percentage_error}")

>>> MAPE: 7.26
>>> CPU times: user 2h 10min 40s, sys: 2min 39s, total: 2h 13min 19s 
>>> Wall time: 35min 15s
```

Then we perform identical cross validation using `update`. The code is copied directly from the [documentation](http://alkaline-ml.com/pmdarima/usecases/stocks.html)
```python
# test 2 
%%time
# source: http://alkaline-ml.com/pmdarima/usecases/stocks.html

model = est1  # seeded from the model we've already fit

def forecast_one_step():
    fc, conf_int = model.predict(n_periods=1, return_conf_int=True)
    return (
        fc.tolist()[0],
        np.asarray(conf_int).tolist()[0])

forecasts = []
confidence_intervals = []

for new_ob in y_test:
    fc, conf = forecast_one_step()
    forecasts.append(fc)
    confidence_intervals.append(conf)

    # Updates the existing model with a small number of MLE steps
    model.update(new_ob)
    print(new_ob, end='\r')

y_pred_2 = pd.Series(forecasts, index=y_test.index)
pr_iu = PredictionResult()
pr_iu.fit(y_pred_2, y_test)

print(f"MAPE: {pr_iu.mean_absolute_percentage_error}")

>>> MAPE: 6.74
>>> CPU times: user 28min 58s, sys: 45.3 s, total: 29min 43s
>>> Wall time: 8min 37s
```

We see that *test 2* resulted in better performance with MAPE of 6.74 % as compared to 7.26 % in *test 1*. Moreover, it took only 8 minutes to run *test 2* as compared to 35 minutes for *test 1*. 

**Conclusion**
Using `update` instead of `fit` seems to lead to better performance  with better efficiency when performing rolling forecast cross validation. There has been some discussion related to differences of `fit` and `update` (see [#256](https://github.com/alkaline-ml/pmdarima/issues/256)) but I couldn't find a thread specifically on this issue. I would love to hear the thoughts from core developers on this. Am I doing something silly, have I misunderstood the API or is this relevant?  Would it make sense to add an option to use `update` insted of `fit` when using cross_val_predict?

Thank you in advance!

*I also want to take the opportunity to express my gratitude for the impressive and enormous effort on the package @tgsmith61591. The result is elegant and the novel cross validation capabilites really make a cherry on top.*

**Versions (if necessary)**

System:
    python: 3.6.7 | packaged by conda-forge | (default, Nov  6 2019, 16:03:31)  [GCC Clang 9.0.0 (tags/RELEASE_900/final)]
executable: /Users/jalmarituominen/miniconda3/envs/covariates/bin/python
   machine: Darwin-18.6.0-x86_64-i386-64bit

Python dependencies:
        pip: 20.0.2
 setuptools: 45.2.0.post20200209
    sklearn: 0.22.1
statsmodels: 0.11.1
      numpy: 1.18.1
      scipy: 1.4.1
     Cython: 0.29.14
     pandas: 1.0.1
     joblib: 0.14.1
   pmdarima: 1.6.1

**Comment by tgsmith61591** [comment]
Hey @tuomijal sorry for the delay in response, and thanks so much for the well formatted question.

The primary difference between `fit` and `update` is that `fit` will perform a full, fresh model fit (no starting params) over the training data (in this case a fold) while `update` will take several new steps with the original training set + the new observations, starting from the model's existing params. I think your idea makes sense, but we'd probably want to do it in a separate method altogether. Currently if we ran `update` rather than `fit` over a fold that overlapped with the previous fold, you'd end up calling `update` over repeated slices of the same training set which is obviously undesirable. We'd probably want to make sure we only update with the newly observed samples, if that makes sense.

I think this approach would be a valuable addition to the package. I'll try to get that in before the next minor release :)

For what it's worth, I opened [an issue](https://github.com/statsmodels/statsmodels/issues/6740) with statsmodels recently about added support for CSS approximation, which would allow us to fit large models much more quickly. Hopefully we can support that soon.
